src/core/io/mpiio/mpiio.cpp

```diff
@@ -446,9 +446,11 @@ void mpi_mpiio_common_read(const char *filename, unsigned fields) {
 
     for (int i = 0; i < nlocalpart; ++i) {
       int blen = boff[i + 1] - boff[i];
-      auto &bl = local_particles[id[i]]->bl;
-      bl.resize(blen);
-      std::copy_n(&bond[boff[i]], blen, bl.begin());
+      if (blen) {
```

Comment:
I'd actually suggest a `if (blen > 0)` here, because `blen` is not a bool and not an int pretending to be.